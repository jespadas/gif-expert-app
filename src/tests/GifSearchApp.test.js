import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifSearchApp } from '../GifSearchApp';

describe('Pruebas en GifSearchApp', () => {

    test('Debe mostrarse correctamente', () => {

        const wrapper = shallow(<GifSearchApp />);

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe mostrar na lista de categorias', () => {

        const categories = ['Dragon Ball', 'Ranma 1/2'];
        const wrapper = shallow(<GifSearchApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })

})
