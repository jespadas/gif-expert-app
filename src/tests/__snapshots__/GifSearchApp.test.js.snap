// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifSearchApp Debe mostrar na lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Ranma 1/2"
      key="Ranma 1/2"
    />
  </ul>
</Fragment>
`;

exports[`Pruebas en GifSearchApp Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
